lib/spack/spack/spec.py

```diff
@@ -572,35 +581,26 @@ def __init__(self, *args):
             self.versions.add(versions)
 
         else:
-            raise TypeError("__init__ takes 1 or 2 arguments. (%d given)" % nargs)
-
-    def _add_versions(self, version_list):
-        # If it already has a non-trivial version list, this is an error
-        if self.versions and self.versions != vn.VersionList(":"):
-            # Note: This may be impossible to reach by the current parser
-            # Keeping it in case the implementation changes.
-            raise MultipleVersionError(
-                "A spec cannot contain multiple version signifiers." " Use a version list instead."
-            )
-        self.versions = vn.VersionList()
-        for version in version_list:
-            self.versions.add(version)
+            raise TypeError(f"__init__ takes 1 or 2 arguments. ({nargs:d} given)")
 
-    def _autospec(self, compiler_spec_like):
+    @staticmethod
+    def _maybe_literal(compiler_spec_like: Union[str, "CompilerSpec"]) -> "CompilerSpec":
+        """Convert a possible string literal to a CompilerSpec object"""
         if isinstance(compiler_spec_like, CompilerSpec):
             return compiler_spec_like
-        return CompilerSpec(compiler_spec_like)
+        spec = spack.parser.parse_one_or_raise(f"%{compiler_spec_like}")
+        return spec.compiler
 
-    def satisfies(self, other, strict=False):
-        other = self._autospec(other)
+    def satisfies(self, other: Union[str, "CompilerSpec"], strict: bool = False) -> bool:
```

Comment:
I think `CompilerSpecLike = Union[str, "CompilerSpec"]` would be useful somewhere -- then we wouldn't have to write this out.